#Config file for Kalman Parameters

#Lugre Kalman Filter
lugre_ekf:

  #Matrix P
  initial_estimate_covariance: [1e-5, 0,    0,    0,    0,    0,    0,     0,     0,    0,    0,    0,    0,    0,    0,
                                0,    1e-5, 0,    0,    0,    0,    0,     0,     0,    0,    0,    0,    0,    0,    0,
                                0,    0,    1e-5, 0,    0,    0,    0,     0,     0,    0,    0,    0,    0,    0,    0,
                                0,    0,    0,    1e-5, 0,    0,    0,     0,     0,    0,    0,    0,    0,    0,    0,
                                0,    0,    0,    0,    1e-5, 0,    0,     0,     0,    0,    0,    0,    0,    0,    0,
                                0,    0,    0,    0,    0,    1e-5, 0,     0,     0,    0,    0,    0,    0,    0,    0,
                                0,    0,    0,    0,    0,    0,    1e-5,  0,     0,    0,    0,    0,    0,    0,    0,
                                0,    0,    0,    0,    0,    0,    0,     1e-5,  0,    0,    0,    0,    0,    0,    0,
                                0,    0,    0,    0,    0,    0,    0,     0,     1e-5, 0,    0,    0,    0,    0,    0,
                                0,    0,    0,    0,    0,    0,    0,     0,     0,    1e-5, 0,    0,    0,    0,    0,
                                0,    0,    0,    0,    0,    0,    0,     0,     0,    0,    1e-5, 0,    0,    0,    0,
                                0,    0,    0,    0,    0,    0,    0,     0,     0,    0,    0,    1e-5, 0,    0,    0,
                                0,    0,    0,    0,    0,    0,    0,     0,     0,    0,    0,    0,    1e-5, 0,    0,
                                0,    0,    0,    0,    0,    0,    0,     0,     0,    0,    0,    0,    0,    1e-5, 0,
                                0,    0,    0,    0,    0,    0,    0,     0,     0,    0,    0,    0,    0,    0,    1e-5]
  
  #Matrix Q
  process_covariance:  [1e-7, 0,    0,    0,    0,    0,    0,     0,     0,    0,    0,    0,    0,    0,    0,
                        0,    1e-7, 0,    0,    0,    0,    0,     0,     0,    0,    0,    0,    0,    0,    0,
                        0,    0,    1e-2, 0,    0,    0,    0,     0,     0,    0,    0,    0,    0,    0,    0,
                        0,    0,    0,    1e-7, 0,    0,    0,     0,     0,    0,    0,    0,    0,    0,    0,
                        0,    0,    0,    0,    1e-7, 0,    0,     0,     0,    0,    0,    0,    0,    0,    0,
                        0,    0,    0,    0,    0,    1e-2, 0,     0,     0,    0,    0,    0,    0,    0,    0,
                        0,    0,    0,    0,    0,    0,    1e-7,  0,     0,    0,    0,    0,    0,    0,    0,
                        0,    0,    0,    0,    0,    0,    0,     1e-7,  0,    0,    0,    0,    0,    0,    0,
                        0,    0,    0,    0,    0,    0,    0,     0,     1e-2, 0,    0,    0,    0,    0,    0,
                        0,    0,    0,    0,    0,    0,    0,     0,     0,    1e-7, 0,    0,    0,    0,    0,
                        0,    0,    0,    0,    0,    0,    0,     0,     0,    0,    1e-7, 0,    0,    0,    0,
                        0,    0,    0,    0,    0,    0,    0,     0,     0,    0,    0,    1e-2, 0,    0,    0,
                        0,    0,    0,    0,    0,    0,    0,     0,     0,    0,    0,    0,    1e-4, 0,    0,
                        0,    0,    0,    0,    0,    0,    0,     0,     0,    0,    0,    0,    0,    1e-4, 0,
                        0,    0,    0,    0,    0,    0,    0,     0,     0,    0,    0,    0,    0,    0,    1e-4]

  #Matrix R
  measurement_covariance:  [0.05, 0,    0,    0,    0,     0,    0,
                            0,    0.05, 0,    0,    0,     0,    0,
                            0,    0,    0.5,  0,    0,     0,    0,
                            0,    0,    0,    0.5,  0,     0,    0,
                            0,    0,    0,    0,    0.001, 0,    0,
                            0,    0,    0,    0,    0,     0.1,  0,
                            0,    0,    0,    0,    0,     0,    1e-4]


#GPS Position Kalman Filter
gps_ekf:

  #If you want to use the gps for correction in the position estimation
  use_gps: false

  #Matrix P
  initial_estimate_covariance: [1e-5, 0,    0,
                                0,    1e-5, 0,
                                0,    0,    1e-5]

  #Matrix Q
  process_covariance:  [1e-2, 0,    0,
                        0,    1e-2, 0,
                        0,    0,    1e-4]
  
  #Matrix R
  measurement_covariance:  [1e-2, 0,    0,
                            0,    1e-2, 0,
                            0,    0,    1e-2]
